File: archived/src/go/ch01/1.3/l0/parity.go
package main

func IsEven(n int) bool { return n%2 == 0 }
func IsOdd(n int) bool  { return n%2 != 0 }

// ParityBit returns "even" or "odd" using the last binary bit.
func ParityBit(n int) string {
	if n&1 == 1 {
		return "odd"
	}
	return "even"
}

File: archived/src/python/ch01/1.3/l0/parity.py
def is_even(n: int) -> bool:
    """Return True if n is even (mod-based)."""
    return n % 2 == 0

def is_odd(n: int) -> bool:
    """Return True if n is odd."""
    return n % 2 != 0

def parity_bit(n: int) -> str:
    """Return 'even' or 'odd' using the last binary bit."""
    return "odd" if (n & 1) else "even"
